internal/tekton/test_client.go

```diff
@@ -0,0 +1,38 @@
+package tekton
+
+import (
+	"context"
+	"fmt"
+
+	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestClient returns mocked pipelines.
+type TestClient struct {
+	// Pipelines is the pool of pipelines which can be retrieved.
+	Pipelines []*tekton.Pipeline
+	// CreatedPipelines is a slice of created pipeline names.
+	CreatedPipelines []string
+	// UpdatedPipelines is a slice of updated pipeline names.
+	UpdatedPipelines []string
```

Comment:
Was wondering if it would make sense to include two `bool`s (e.g. `FailUpdatePipeline` and `FailCreatePipeline`) that allow us to let `CreatePipeline` or `UpdatePipeline` fail/return non-nil errors? That way, we could excercise error handling logic in the unit tests as well.

Reply:
Good point. Added in 7029a40.